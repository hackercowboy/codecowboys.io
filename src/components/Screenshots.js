import React from 'react';
import ImageGallery from 'react-image-gallery';
import PropTypes from 'prop-types';

import './Screenshots.scss';

function Screenshots({ ipadImages, iphoneImages }) {
  return (
    <div className="screenshots">
      <div className="mobile">
        <ImageGallery items={iphoneImages} />
      </div>
      <div className="desktop">
        <ImageGallery items={ipadImages} />
      </div>
    </div>
  );
}

Screenshots.propTypes = {
  ipadImages: PropTypes.arrayOf(PropTypes.shape({
    original: PropTypes.string.isRequired,
    thumbnail: PropTypes.string.isRequired,
  })).isRequired,
  iphoneImages: PropTypes.arrayOf(PropTypes.shape({
    original: PropTypes.string.isRequired,
    thumbnail: PropTypes.string.isRequired,
  })).isRequired,
};

export default Screenshots;
