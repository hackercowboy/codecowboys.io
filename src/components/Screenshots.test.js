import React from 'react';
import { shallow } from 'enzyme';

import Screenshots from './Screenshots';

describe('<Screenshots/>', () => {
  it('should render', () => {
    const ipadImages = [1, 2, 3, 4, 5].map((index) => ({
      original: `/screenshots/letter/letter_ipad_${index}_de.png`,
      thumbnail: `/screenshots/letter/letter_ipad_${index}_de.png`,
    }));

    const iphoneImages = [1, 2, 3, 4, 5].map((index) => ({
      original: `/screenshots/letter/letter_iphone_${index}_de.png`,
      thumbnail: `/screenshots/letter/letter_iphone_${index}_de.png`,
    }));
    const wrapper = shallow(<Screenshots ipadImages={ipadImages} iphoneImages={iphoneImages} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
