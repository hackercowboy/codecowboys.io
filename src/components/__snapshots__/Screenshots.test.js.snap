// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Screenshots/> should render 1`] = `
"<div className=\\"screenshots\\">
  <div className=\\"mobile\\">
    <l items={{...}} onErrorImageURL=\\"\\" additionalClass=\\"\\" showNav={true} autoPlay={false} lazyLoad={false} infinite={true} showIndex={false} showBullets={false} showThumbnails={true} showPlayButton={true} showFullscreenButton={true} disableThumbnailScroll={false} disableKeyDown={false} disableSwipe={false} disableThumbnailSwipe={false} useTranslate3D={true} isRTL={false} useBrowserFullscreen={true} flickThreshold={0.4} stopPropagation={false} indexSeparator=\\" / \\" thumbnailPosition=\\"bottom\\" startIndex={0} slideDuration={450} swipingTransitionDuration={0} swipingThumbnailTransitionDuration={0} onSlide={{...}} onBeforeSlide={{...}} onScreenChange={{...}} onPause={{...}} onPlay={{...}} onClick={{...}} onImageLoad={{...}} onImageError={{...}} onTouchMove={{...}} onTouchEnd={{...}} onTouchStart={{...}} onMouseOver={{...}} onMouseLeave={{...}} onThumbnailError={{...}} onThumbnailClick={{...}} renderCustomControls={{...}} renderThumbInner={{...}} renderItem={{...}} slideInterval={3000} slideOnThumbnailOver={false} swipeThreshold={30} renderLeftNav={[Function: renderLeftNav]} renderRightNav={[Function: renderRightNav]} renderPlayPauseButton={[Function: renderPlayPauseButton]} renderFullscreenButton={[Function: renderFullscreenButton]} useWindowKeyDown={true} />
  </div>
  <div className=\\"desktop\\">
    <l items={{...}} onErrorImageURL=\\"\\" additionalClass=\\"\\" showNav={true} autoPlay={false} lazyLoad={false} infinite={true} showIndex={false} showBullets={false} showThumbnails={true} showPlayButton={true} showFullscreenButton={true} disableThumbnailScroll={false} disableKeyDown={false} disableSwipe={false} disableThumbnailSwipe={false} useTranslate3D={true} isRTL={false} useBrowserFullscreen={true} flickThreshold={0.4} stopPropagation={false} indexSeparator=\\" / \\" thumbnailPosition=\\"bottom\\" startIndex={0} slideDuration={450} swipingTransitionDuration={0} swipingThumbnailTransitionDuration={0} onSlide={{...}} onBeforeSlide={{...}} onScreenChange={{...}} onPause={{...}} onPlay={{...}} onClick={{...}} onImageLoad={{...}} onImageError={{...}} onTouchMove={{...}} onTouchEnd={{...}} onTouchStart={{...}} onMouseOver={{...}} onMouseLeave={{...}} onThumbnailError={{...}} onThumbnailClick={{...}} renderCustomControls={{...}} renderThumbInner={{...}} renderItem={{...}} slideInterval={3000} slideOnThumbnailOver={false} swipeThreshold={30} renderLeftNav={[Function: renderLeftNav]} renderRightNav={[Function: renderRightNav]} renderPlayPauseButton={[Function: renderPlayPauseButton]} renderFullscreenButton={[Function: renderFullscreenButton]} useWindowKeyDown={true} />
  </div>
</div>"
`;
